rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -444,39 +444,34 @@ struct CallbackHandler {
     resources: Arc<ArcSwap<Vec<ResourceDescription>>>,
 }
 
-#[derive(thiserror::Error, Debug)]
-enum CallbackError {
-    #[error("system DNS resolver problem: {0}")]
-    Resolvers(#[from] client::resolvers::Error),
-    #[error("can't send to controller task: {0}")]
-    SendError(#[from] mpsc::error::TrySendError<ControllerRequest>),
-}
-
 // Callbacks must all be non-blocking
 impl connlib_client_shared::Callbacks for CallbackHandler {
-    type Error = CallbackError;
-
-    fn on_disconnect(&self, error: &connlib_client_shared::Error) -> Result<(), Self::Error> {
+    fn on_disconnect(&self, error: &connlib_client_shared::Error) {
         tracing::debug!("on_disconnect {error:?}");
-        self.ctlr_tx.try_send(ControllerRequest::Disconnected)?;
-        Ok(())
+        let _ = self.ctlr_tx.try_send(ControllerRequest::Disconnected);
```

Comment:
```suggestion
        let _ = self.ctlr_tx.try_send(ControllerRequest::Disconnected).expect("Should be able to send messages to `Controller`");
```

If there's no error handling then it should panic instead of desyncing the GUI.